src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Product/Form/customizations_form_theme.html.twig

```diff
@@ -0,0 +1,39 @@
+{%- block customizations_widget -%}
+  <div class="col-md-12">
+    <h2>{{ 'Customization'|trans({}, 'Admin.Catalog.Feature') }}</h2>
+    <ul class="list-unstyled" id="customization-fields-list" data-prototype="{{ form_row(form.customization_fields.vars.prototype)|e('html_attr') }}">
+      {{ form_row(form.customization_fields) }}
+    </ul>
+    <button type="button" class="btn btn-outline-secondary" id="add-customization-btn">
+      <i class="material-icons">add_circle</i>
+      {{ 'Add a customization field'|trans({}, 'Admin.Catalog.Feature') }}
+    </button>
+  </div>
+{%- endblock customizations_widget -%}
+
+{%- block customization_field_row -%}
+  <li class="form-group row customization-field-item">
+    {{ form_widget(form.id) }}
+    <div class="col-md-5">
+      {{ form_widget(form.name) }}
+      {{ form_errors(form.name) }}
+    </div>
+    <div class="col-md-4">
+      {{ form_widget(form.type) }}
+      {{ form_errors(form.type) }}
+    </div>
+    <div class="col-md-1">
+      <button type="button" class="btn text-secondary remove-customization-btn">
+        <i class="material-icons">delete</i>
+      </button>
+    </div>
+    <div class="col-md-2">
+      {{ form_widget(form.required) }}
+      {{ form_errors(form.required) }}
+    </div>
+  </li>
+{%- endblock customization_field_row -%}
+
+{%- block _product_customizations_customization_fields_row -%}
```

Comment:
I believe, not sure though, this is the block which should integrate your custom design (along with the H2) (based on how Symfony creates its own theme) The row is all that surrounds the widget itself (label, errors, ...), the widget should only be the `ul` list itself

This way in the product page template you can simply use `{{ form_row(productForm.customizations) }}` (not `form_widget`)